src/main.rs

```diff
@@ -131,7 +131,17 @@ fn try_main() -> Result<()> {
             bail!("not found release note for '{}'", version);
         }
     } else {
-        &changelog[0]
+        let (entry_key, entry_value) = changelog.first().context("not found release")?;
+
+        if entry_key != &"Unreleased" {
+            entry_value
+        } else {
+            (
+                changelog.get_index(1).context(
```

Comment:
It is nice to use with_context to call format! only when necessary.

```suggestion
                changelog.get_index(1).with_context(||
```

Reply:
Isn't format! called regardless at build time since it's a macro?

Reply:
`format!` is a macro for formatting strings *at runtime*.
There is also a macro called `concat!` that concatenates string literals at compile time, but it cannot be used in this use case.

Reply:
That makes sense since the variables aren't known until runtime, I guess I hadn't thought about it much before. I'm seeing how that'd cause less overhead - does that closure just not get called unless it's shown to be needed then (compared to format always being called)?